crypto/asn1/asn1_lib.c

```diff
@@ -126,12 +125,12 @@ static int asn1_get_length(const unsigned char **pp, int *inf, long *rl,
                 p++;
                 i--;
             }
-            if (i > sizeof(long))
+            if (i > sizeof(long) || i == 0)
                 return 0;
-            while (i-- > 0) {
+            do {
                 ret <<= 8L;
```

Comment:
BTW, wouldn't it be also appropriate to freshen up previous loop? Some would argue that `while (i && ...)` should be explicit `while (i != 0 && ...)`. In the context one can also wonder if it would be appropriate if they were more alike? I.e. both `while (i != 0 ...` with `i--` in loop body? This is basically side comment, i.e. whether or not you actually choose to do something is up to you.